api/server/httprest/server.go

```diff
@@ -48,7 +49,7 @@ func New(ctx context.Context, opts ...Option) (*Server, error) {
 		defaultReadHeaderTimeout = g.cfg.timeout
 		handler = http.TimeoutHandler(g.cfg.router, g.cfg.timeout, "request timed out")
 	} else {
-		handler = g.cfg.router
+		handler = middleware.MiddlewareChain(g.cfg.router, g.cfg.middlewares)
```

Comment:
middlewares are not being passed as options in validator/rpc/server.go they are not registered